### ISSUE TITLE
[typescript] Broken compatibility with @types/next

### ISSUE THREAD

**Issue by adam187** [zeit/next.js#7247]
# Bug report

## Describe the bug

Looks like this after this change https://github.com/zeit/next.js/pull/7154/files#diff-e5f184d6408bb983da44fe3b0f27d4faR10 external types from [@types/next](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/next)
are no longer used.

## To Reproduce

```bash
git clone git@github.com:adam187/next-js-ts-error-example.git
yarn install
yarn ts:check:server
```

## Expected behavior
Types should work properly 

## System information
 - Version of Next.js: 8.1.1-canary.12

**Comment by timneutkens** [comment]
Doesn't seem like a bug to me. Why would you want to use externally managed types in the `@types` namespace when the official types are literally generated from the source code that you install.

**Comment by adam187** [comment]
@timneutkens I agree it's not really a bug, but it can be problematic for typescript users that want to upgrade to `next.js@8.1.1` and they used to use `@types/next` for declarations. 

Problem is that https://github.com/zeit/next.js/blob/canary/packages/next/types/index.d.ts is missing some declarations that https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/next/index.d.ts have.

Basically, I'm just reporting an issue that I occurred while upgrading.

**Comment by Timer** [comment]
Hi @adam187! Unfortunately, this is unavoidable for any package transitioning from external typings to provided type declarations.

I'll close this as unactionable -- btw: we'll likely release these changes under `8.2.0` (not `8.1.1`)!

**Comment by nhooyr** [comment]
@timneutkens could you please direct me to where the official types are? I cannot seem to find any documentation on them.

**Comment by nhooyr** [comment]
Ah the latest version is 8.1.0 which I think does not include the types.

**Comment by Timer** [comment]
@nhooyr you can install `next@canary` to try them!

**Comment by nhooyr** [comment]
They work wonderfully, thanks!

**Comment by nhooyr** [comment]
@Timer only issue I'm having with is styled-jsx doesn't seem to be supported. I'm using canary next and styled-jsx. Any ideas?

**Comment by Timer** [comment]
Please file an issue @nhooyr it's hard to track things in closed issues. 🙏
